web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -135,9 +113,22 @@ export class LayerDialogComponent implements OnDestroy {
   onProjectLoaded(project: Project) {
     if (this.layerId === ':new') {
       this.layerId = this.dataStoreService.generateId();
+      const fieldId = this.dataStoreService.generateId();
```

Comment:
If we update the component to accept `Layer?` instead of Project and Id, a null `Layer` could mean "new" here. This also makes sense since `:new` is a detail of the URL/router that the component shouldn't know/care about.